opentelemetry-exporter-gcp-logging/src/opentelemetry/exporter/cloud_logging/__init__.py

```diff
@@ -0,0 +1,226 @@
+        log_entries = []
+        for log_data in batch:
+            log_record = log_data.log_record
+            attributes = log_record.attributes or {}
+            project_id = self.project_id
+            if attributes.get(PROJECT_ID_ATTRIBUTE_KEY):
+                project_id = str(attributes.get(PROJECT_ID_ATTRIBUTE_KEY))
+            log_name = self.default_log_name
+            if attributes.get(LOG_NAME_ATTRIBUTE_KEY):
+                log_name = str(attributes.get(LOG_NAME_ATTRIBUTE_KEY))
+            monitored_resource_data = get_monitored_resource(
+                log_record.resource or Resource({})
+            )
+            monitored_resource_data = get_monitored_resource(
+                log_record.resource or Resource({})
+            )
+            # convert it to proto
+            monitored_resource: Optional[MonitoredResource] = (
+                MonitoredResource(
+                    type=monitored_resource_data.type,
+                    labels=monitored_resource_data.labels,
+                )
+                if monitored_resource_data
+                else None
+            )
+            # if timestamp is unset, fall back to observed_time_unix_nano as recommended
+            #   (see https://github.com/open-telemetry/opentelemetry-proto/blob/4abbb78/opentelemetry/proto/logs/v1/logs.proto#L176-L179)
+            ts = Timestamp()
+            if log_record.timestamp or log_record.observed_timestamp:
+                ts.FromNanoseconds(
+                    log_record.timestamp or log_record.observed_timestamp
+                )
+            else:
+                ts.FromDatetime(datetime.datetime.now())
+            log_name = "projects/{}/logs/{}".format(
+                project_id, urllib.parse.quote_plus(log_name)
+            )
+            log_entry = LogEntry()
+            log_entry.timestamp = ts
+            log_entry.log_name = log_name
+            if monitored_resource:
+                log_entry.resource = monitored_resource
+            if monitored_resource:
+                log_entry.resource = monitored_resource
+            attrs_map = {k: v for k, v in attributes.items()}
+            log_entry.trace_sampled = (
+                log_record.trace_flags is not None
+                and log_record.trace_flags.sampled
+            )
+            if TRACE_SAMPLED_ATTRIBUTE_KEY in attrs_map:
+                log_entry.trace_sampled |= bool(
+                    attrs_map[TRACE_SAMPLED_ATTRIBUTE_KEY]
+                )
+                log_entry.trace_sampled |= bool(
+                    attrs_map[TRACE_SAMPLED_ATTRIBUTE_KEY]
+                )
+                del attrs_map[TRACE_SAMPLED_ATTRIBUTE_KEY]
+            if log_record.trace_id:
+                log_entry.trace = "projects/{}/traces/{}".format(
+                    project_id, log_record.trace_id
+                )
+            if log_record.span_id:
+                log_entry.span_id = str(hex(log_record.span_id))[2:]
```

Comment:
There's a util you can use for this https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/blob/29e3d49ad76060086a86391260665ebfa95ad641/opentelemetry-exporter-gcp-trace/src/opentelemetry/exporter/cloud_trace/__init__.py#L119